datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -2708,379 +2708,428 @@ mod tests {
         ))
     }
 
-    #[tokio::test]
-    async fn test_union_equivalence_properties_multi_children() -> Result<()> {
-        let schema = create_test_schema()?;
+    #[test]
+    fn test_union_equivalence_properties_multi_children_1() {
```

Comment:
The individual functions corresponds to TEST CASE N. 

I think they are easier to understand, are more concise and run in parallel (and I can run individual tests)

Reply:
After the refactor, it looks great.